packages/internals/src/engine-commands/getBinaryVersion.ts

```diff
@@ -0,0 +1,37 @@
+import type { NodeAPILibraryTypes } from '@prisma/engine-core'
+import { getCliQueryEngineBinaryType } from '@prisma/engines'
+import { BinaryType } from '@prisma/fetch-engine'
+import { isNodeAPISupported } from '@prisma/get-platform'
+import execa from 'execa'
+import * as TE from 'fp-ts/TaskEither'
+
+import { resolveBinary } from '../resolveBinary'
+import { load } from '../utils/load'
+
+const MAX_BUFFER = 1_000_000_000
+
+export async function getBinaryVersion(enginePath?: string, binaryName?: BinaryType): Promise<string> {
+  if (!binaryName) {
+    binaryName = getCliQueryEngineBinaryType()
+  }
+  enginePath = await resolveBinary(binaryName, enginePath)
+  if (binaryName === BinaryType.libqueryEngine) {
+    await isNodeAPISupported()
+
+    const QE = load<NodeAPILibraryTypes.Library>(enginePath)
+    return `libquery-engine ${QE.version().commit}`
+  } else {
+    const result = await execa(enginePath, ['--version'], {
+      maxBuffer: MAX_BUFFER,
```

Comment:
`MAX_BUFFER` should not be needed here, the engine output will be small (for getDMMF it can be very large and then it's needed there)
The default is `Default: 100_000_000 (100 MB)` https://github.com/sindresorhus/execa#maxbuffer

Reply:
Thank you, please notice I didn't add `MAX_BUFFER` here. Should we change this in another PR, after this gets merged?

Reply:
Actually, this is small enough we can do it in this PR